### PR TITLE
Differentiate base paths in repository integration tests

### LINKED ISSUE #47202: [CI] AzureStorageCleanupThirdPartyTests.testCreateSnapshot failuire
Example failure:

https://gradle-enterprise.elastic.co/s/zb2xff2k72smu/console-log#L926

same build as in #47201, could be related, looking at the build-stats it also started failing yesterday (Sept 26th):

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-90d,mode:quick,to:now))&_a=(columns:!(_source),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:AzureStorageCleanupThirdPartyTests),sort:!(time,desc))

Failure:

```
org.elasticsearch.repositories.RepositoryException: [test-repo] concurrent modification of the index-N file, expected current generation [0], actual current generation [1] - possibly due to simultaneous snapshot deletion requests
```

Trying to reproduce locally I get some other failure.

### PR DESCRIPTION
**Pull request by tlrx**
This commit change the repositories base paths used in Azure integration tests so that they don't conflict with each other when tests run in parallel on real Azure storage service.

Closes #47202

### CODE REVIEW

File: plugins/repository-azure/build.gradle
Comment (original-brownbear) [inline plugins/repository-azure/build.gradle]: @tlrx thanks for this one!

I think technically this is the perfect fix in terms of stability. One problem we're creating here though is that we are leaking a bunch of blobs for each test run now.
Should we add some logic to delete the leaked folders maybe? (otherwise we force infra to deal with this somehow which will be tricky for them I think)

Comment (original-brownbear) [reply plugins/repository-azure/build.gradle]: To avoid this issue for now: Maybe we can simply add a constant path here and another constant in the other build so we don't get interference between the test runs? We don't have to worry about different builds interfering with each other, infra makes sure that doesn't happen so that should be fine?

Comment (tlrx) [reply plugins/repository-azure/build.gradle]: Yeah, I wrongly assumed that the container/buckets were recreated on demand by some Infra task before running the tests.

I think we can easily keep the base_path with test seed for the 3rd party tests, as we can easily clean up the files when tearing down the test (similarly to what the test set up is already doing doing)

For the repository integration test, I'll go with adding a constant.

**Review by original-brownbear — APPROVED** [review]
LGTM (assuming CI is happy too) :) Its retriever the 3rd party tests right away after merging this to make sure we're good here.

**Review by original-brownbear — CHANGES_REQUESTED** [review]
Seems CI is unhappy, that teardown doesn't work with HDFS repo for some reason.

**Comment by tlrx** [comment]
@elasticmachine update branch

**Comment by tlrx** [comment]
@original-brownbear I've updated to code to fix the HDFS test issue. I've also added changes for S3/GCS (will update title+desc when merging) alongside. Can you please have another look?

**Review by original-brownbear — APPROVED** [review]
LGTM (if CI loves it :D), thanks @tlrx !

**Comment by tlrx** [comment]
Thanks Armin!

**Comment by romseygeek** [comment]
This just failed on 6.8 as well, can it be backported there?

https://gradle-enterprise.elastic.co/s/pdvqg6hefhjd6/

**Comment by tlrx** [comment]
Thanks @romseygeek - I opened #48286

**Pull request merged**
